File: _site/technical/leet/ransomNote.py
# 383
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        magCounter = Counter(magazine)
        for letter in ransomNote:
            if magCounter[letter] < 1:
                return(False)
            
            magCounter[letter] -= 1
        
        return(True)

File: docs/technical writing/leet/ransomNote.cpp
#include <vector>;
#include <iostream>;
using namespace std;
//383
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> mag;
        unordered_map<char, int> ransom;
        
        for(char r: ransomNote) {
            ransom[r] += 1;
            
        }
        for(char m: magazine) {
            mag[m] += 1;
            
        }
        for(auto [key, val] : ransom) {
            if(val > mag[key]) {
                return(false);
            }
        }
        return(true);
        
    }
};